cicd/devnet/terraform/main.tf

```diff
@@ -9,114 +9,97 @@ terraform {
   required_version = ">= 1.2.0"
 }
 
+# Default
 provider "aws" {
   region  = "us-east-1"
 }
 
-resource "aws_vpc" "devnet_vpc" {
-  cidr_block = "10.0.0.0/16"
-  instance_tenancy = "default"
-  enable_dns_hostnames = true
-  
-  tags = {
-    Name = "TfDevnetVpc"
-  }
-}
-
-resource "aws_subnet" "devnet_subnet" {
-  vpc_id = aws_vpc.devnet_vpc.id
-  cidr_block = "10.0.0.0/20"
-  map_public_ip_on_launch = true
-  availability_zone = "us-east-1a"
-  
-  tags = {
-    Name = "TfDevnetVpcSubnet"
-  }
+provider "aws" {
+  alias = "us-east-2"
+  region  = "us-east-2"
 }
 
-resource "aws_internet_gateway" "devnet_gatewat" {
-  vpc_id = aws_vpc.devnet_vpc.id
+module "us-east-2" {
+  source = "./module/region"
+  availability_zone = "us-east-2a"
+  region = "us-east-2"
+  devnetNodeKeys = local.devnetNodeKeys["us-east-2"]
+  logLevel = local.logLevel
+  devnet_xdc_ecs_tasks_execution_role_arn = aws_iam_role.devnet_xdc_ecs_tasks_execution_role.arn
 
-  tags = {
-    Name = "TfDevnetGateway"
+  providers = {
+    aws = aws.us-east-2
   }
 }
 
-resource "aws_route_table" "devnet_route_table" {
-  vpc_id = aws_vpc.devnet_vpc.id
+provider "aws" {
+  alias = "eu-west-1"
+  region  = "eu-west-1"
+}
 
-  route {
-    cidr_block = "0.0.0.0/0"
-    gateway_id = aws_internet_gateway.devnet_gatewat.id
-  }
+module "eu-west-1" {
+  source = "./module/region"
+  availability_zone = "eu-west-1a"
+  region = "eu-west-1"
+  devnetNodeKeys = local.devnetNodeKeys["eu-west-1"]
+  logLevel = local.logLevel
+  devnet_xdc_ecs_tasks_execution_role_arn = aws_iam_role.devnet_xdc_ecs_tasks_execution_role.arn
 
-  tags = {
-    Name = "TfDevnetVpcRoutingTable"
+  providers = {
+    aws = aws.eu-west-1
   }
 }
 
-resource "aws_route_table_association" "devnet_route_table_association" {
-  subnet_id      = aws_subnet.devnet_subnet.id
-  route_table_id = aws_route_table.devnet_route_table.id
+provider "aws" {
+  alias = "sa-east-1"
+  region  = "sa-east-1"
 }
 
-resource "aws_default_security_group" "devnet_xdcnode_security_group" {
-  vpc_id = aws_vpc.devnet_vpc.id
+module "sa-east-1" {
+  source = "./module/region"
+  availability_zone = "sa-east-1a"
+  region = "sa-east-1"
+  devnetNodeKeys = local.devnetNodeKeys["sa-east-1"]
+  logLevel = local.logLevel
+  devnet_xdc_ecs_tasks_execution_role_arn = aws_iam_role.devnet_xdc_ecs_tasks_execution_role.arn
 
-  ingress {
-    from_port   = 0
-    to_port     = 0
-    protocol    = -1
-    cidr_blocks = ["0.0.0.0/0"]
+  providers = {
+    aws = aws.sa-east-1
   }
+}
 
-  egress {
-    from_port   = 0
-    to_port     = 0
-    protocol    = "-1"
-    cidr_blocks = ["0.0.0.0/0"]
-  }
-  tags = {
-    Name = "TfDevnetNode"
-  }
+provider "aws" {
+  alias = "ap-northeast-1"
+  region  = "ap-northeast-1"
 }
 
-# IAM policies
-data "aws_iam_policy_document" "xdc_ecs_tasks_execution_role" {
-  statement {
-    actions = ["sts:AssumeRole"]
+module "ap-northeast-1" {
+  source = "./module/region"
+  availability_zone = "ap-northeast-1a"
```

Comment:
why need to specify the AZ?

Reply:
Initially I thought this is necessary parameter, but yeah, this is optional, I will remove it for neat code.
```
resource "aws_subnet" "devnet_subnet" {
    availability_zone = "us-east-1a"
}
```